graphql/resolvers.go

```diff
@@ -1819,29 +1819,65 @@ func (r *mutationResolver) UnschedulePatchTasks(ctx context.Context, patchID str
 	return &patchID, nil
 }
 
-// We only allow scheduling undispatched base tasks for tasks that are failing on the current patch
+// ScheduleUndispatchedBaseTasks only allows scheduling undispatched base tasks for tasks that are failing on the current patch
 func (r *mutationResolver) ScheduleUndispatchedBaseTasks(ctx context.Context, patchID string) ([]*restModel.APITask, error) {
 	opts := data.TaskFilterOptions{
 		Statuses:     evergreen.TaskFailureStatuses,
 		BaseStatuses: []string{evergreen.TaskUndispatched},
 	}
 	tasks, _, err := r.sc.FindTasksByVersion(patchID, opts)
 	if err != nil {
-		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Could not fetch tasks for patch :%s ", err.Error()))
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Could not fetch tasks for patch: %s ", err.Error()))
 	}
 
 	restartedTasks := []*restModel.APITask{}
+	generatedTasksToRestart := []*task.Task{}
+
 	for _, t := range tasks {
-		// Can ignore error while fetching tasks this could just mean the task didn't exist on the base commit
+		// If a task is a generator task or generated don't schedule it until we get all of the generated tasks we want to generate
+		if !t.GenerateTask && t.GeneratedBy == "" {
```

Comment:
I don't think you need an if statement on this case; even if the task is a generated task, it's possible that the generator already ran on the base commit, in which case the base commit will already have the generated task, so checking FindTaskOnBaseCommit would be reasonable. We only need to keep track of generatedTasksToRestart (and this should probably be generatedTasksToSchedule) if we don't find the generated task on the base commit.

i.e. you're missing these two cases:
1. my patch has lint-model and we want to schedule the base commit's, and the base commit also has generate-lint and lint-model, we should schedule lint-model, not generate-lint.
2. my patch has generate-lint and it failed, then we want to schedule generate-lint on the base commit.

Reply:
We're still missing the second case I believe. We should still consider scheduling generator tasks.